File: app/src/main/java/com/qwert2603/vkmessagestat/VkMessageStatApplication.java
package com.qwert2603.vkmessagestat;

import android.app.Application;

import com.qwert2603.vkmessagestat.di.AppComponent;
import com.qwert2603.vkmessagestat.di.AppModule;
import com.qwert2603.vkmessagestat.di.DaggerAppComponent;
import com.vk.sdk.VKSdk;

public class VkMessageStatApplication extends Application {

    private static AppComponent sAppComponent;

    public static AppComponent getAppComponent() {
        return sAppComponent;
    }

    @Override
    public void onCreate() {
        super.onCreate();

        sAppComponent = buildAppComponent();

        VKSdk.initialize(VkMessageStatApplication.this);
        /*for (String s : VKUtil.getCertificateFingerprint(VkMessageStatApplication.this,
                VkMessageStatApplication.this.getPackageName())) {
            LogUtils.d("CertificateFingerprint", "CertificateFingerprint == " + s);
        }*/
    }

    protected AppComponent buildAppComponent() {
        return DaggerAppComponent.builder()
                .appModule(new AppModule(VkMessageStatApplication.this))
                .build();
    }
}

File: app/src/main/java/com/qwert2603/vkmessagestat/VkMessageStatApplication.kt
package com.qwert2603.vkmessagestat

import android.app.Application
import com.qwert2603.vkmessagestat.di.AppComponent
import com.qwert2603.vkmessagestat.di.AppModule
import com.qwert2603.vkmessagestat.di.DaggerAppComponent
import com.vk.sdk.VKSdk

class VkMessageStatApplication : Application() {

    companion object {
        lateinit var sAppComponent: AppComponent

        fun getAppComponent() = sAppComponent
    }


    override fun onCreate() {
        super.onCreate()

        sAppComponent = buildAppComponent()

        VKSdk.customInitialize(this, resources.getInteger(R.integer.com_vk_sdk_AppId), "5.85")

//        LogUtils.d(VKAccessToken.currentToken().accessToken)

//        for (s in VKUtil.getCertificateFingerprint(this, this.packageName)) {
//            LogUtils.d("CertificateFingerprint", "CertificateFingerprint == $s");
//        }
    }

    fun buildAppComponent(): AppComponent =
            DaggerAppComponent.builder()
                    .appModule(AppModule(this@VkMessageStatApplication))
                    .build()
}